### PR TITLE
fix: collateral share restriction works during partial price outages

### PR DESCRIPTION
**Pull request by toteki**
When some token prices are missing from the oracle, it becomes impossible to calculate a token's collateral share.

However, if the collateralized token's price is known, and enough other token prices are known to prove that it cannot possibly be over its `MaxCollateralShare`, then we can let `MsgSupplyCollateral` or `MsgCollateralize` succeed anyway.

### CODE REVIEW

File: x/leverage/keeper/collateral.go [hunk 1]
```diff
@@ -78,6 +78,33 @@ func (k Keeper) CalculateCollateralValue(ctx sdk.Context, collateral sdk.Coins)
...
+		// convert uToken collateral to base assets
+		baseAsset, err := k.ExchangeUToken(ctx, coin)
+		if err != nil {
+			continue
+		}
+
+		// get USD value of base assets
+		v, err := k.TokenValue(ctx, baseAsset, types.PriceModeSpot)
+		if err != nil {
+			continue
```
Comment (robert-zaremba) [inline x/leverage/keeper/collateral.go:98]: let's log this error with an appropriate message.

File: x/leverage/keeper/collateral.go [hunk 2]
```diff
@@ -112,17 +139,16 @@ func (k Keeper) CollateralLiquidity(ctx sdk.Context, denom string) sdk.Dec {
...
 }
 
-// CollateralShare calculates the portion of overall collateral
-// (measured in USD value) that a given uToken denom represents.
-func (k *Keeper) CollateralShare(ctx sdk.Context, denom string) (sdk.Dec, error) {
+// VisibleCollateralShare calculates the portion of overall collateral (measured in USD value) that a
+// given uToken denom represents. If an asset other than the denom requested is missing an oracle
+// price, it ignores that asset's contribution to the system's overall collateral, thus potentially
+// overestimating the requested denom's collateral share while improving availability.
+func (k *Keeper) VisibleCollateralShare(ctx sdk.Context, denom string) (sdk.Dec, error) {
```
Comment (robert-zaremba) [inline x/leverage/keeper/collateral.go:146]: Do we need to rename this function? I would prefer to not rename it if possible.

**Review by robert-zaremba — APPROVED** [review]
pre-approving.

File: x/leverage/keeper/collateral.go (hunk 1 above)
Comment (toteki) [reply x/leverage/keeper/collateral.go:103]: Added logging at info level so it doesn't flood the error level with noise

**Pull request merged**